source/class/qx/tool/cli/commands/Compile.js

```diff
@@ -1310,6 +1311,42 @@ Framework: v${await this.getQxVersion()} in ${await this.getQxPath()}`);
           (targetConfig.defaultAppConfig.type || "browser") === "browser"
         ) {
           targetConfig.defaultAppConfig.app.setWriteIndexHtmlToRoot(true);
+
+          // Merge the application's (common to targets) commonjs
+          // modules list and the current target's commonjs modules
+          // list
+          commonjsModules =
+            mergeArray(
+              [],
+              targetConfig.defaultAppConfig["commonjs-modules"] || [],
+              targetConfig["commonjs-modules"] || []);
+
+          // If there are any commonjs modules, browserify them
+          //
+          // TODO: Move this to someplace where it can be generated
+          // directly in the output directory. There's no need,
+          // really, for it to be placed in the source directory.
+          if (commonjsModules.length > 0) {
+            let command;
+            let output;
+            let analyser = maker.getAnalyser();
+            const childProcess = require("child_process");
+            command =
+              `browserify -r ${commonjsModules.join(" -r ")}` +
+              ` -o source/resource/script/qx-commonjs-browserified.js`;
+            fs.mkdirSync("source/resource/script", { recursive : true });
+            try {
+              output = childProcess.execSync(command);
+            } catch (e) {
+              qx.tool.compiler.Console.log(`Failed: ${output}`);
+              throw e;
+            }
+
```

Comment:
No compile / build tasks should be done here.  `qx.tool.cli.*` is all about configuring the compiler API, and it is definitely not intended that compiler functionality should be baked into the CLI.  

The `commonjs-modules` in the **target** definition in `compile.json` should be copied into the `qx.tool.compiler.targets.Target` as a property; the `commonjs-modules` in the **application** definition should be copied into `qx.tool.compiler.app.Application` as a property.

As explained in Gitter, in order to include the `commonjs-modules` (which as you say, would be a union of those listed in the target and the application), you need to write an implementation of `qx.tool.compiler.targets.meta.AbstractJavascriptMeta` which would perform the code above which runs browserify.

Once you have a meta class, you need to modify `qx.tool.compiler.targets.Target.generateApplication` to optionally include the new meta class `if (application.getType() == “browser") { …`

Reply:
There's a key question in your gitter that I missed previoously -

>Derrell Lipman @derrell Mar 05 17:35
(Actually, maybe you'd want to add the script as an "external" only if browser is the only application type? I think otherwise, a NodeJS app would get the browserified file...???

the "external" is really used for adding external scripts in the `source/resources` section which are provided by the user; they are expected to be loaded by the boot loader as either resources or from a URL.  

To answer your question about how to prevent the nodejs app getting the browserified file, that's what the `qx.tool.compiler.targets.Target` figures out, and in order to make it so that the browserified code can be easily stacked and used, the `qx.tool.compiler.targets.meta` classes are used.

I'll add a better explanation about why `.meta.` is important below